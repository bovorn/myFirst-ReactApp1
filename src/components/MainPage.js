import React, { Component } from 'react';

import CardList from './CardList';
import ErrorBoundry from './ErrorBoundry';
import SearchBox from './SearchBox';
import Scroll from './Scroll';
import Loading from './Loading';
import Header from './Header';

import './MainPage.css';


class MainPage extends Component {

  componentDidMount() {
    this.props.onRequestRobots();
  }

  filteredRobots = () => {
    return this.props.robots.filter(robot => {
      return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase());
    })
  }

  render() {
    const { onSearchChange, isPending } = this.props;

    return (
      <div className='tc'>
        <div>
        <Header />
        <SearchBox onSearchChange={onSearchChange}/>
        <Scroll isPending={isPending}>
          { isPending ? <Loading /> :
            <ErrorBoundry>
              <CardList robots={this.filteredRobots()} />
            </ErrorBoundry>
          }
        </Scroll>
        </div>
      </div>
    );
  }
}



export default MainPage;
