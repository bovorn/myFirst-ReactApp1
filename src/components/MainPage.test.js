import React from 'react';
import MainPage from './MainPage';
import renderer from 'react-test-renderer';

let wrapper;

beforeEach(() => {
    const mockProps = {
        onRequestRobots: jest.fn(),
        robots: [],
        searchField: '',
        isPending: false
    }
    wrapper = renderer.create(<MainPage {...mockProps} />);
})

it('renders MainPage without crashing', () => {
    expect(wrapper.toJSON()).toMatchSnapshot();
})

it('filters robots correctly', () => {

    expect(wrapper.getInstance().filteredRobots()).toEqual([]);

    const mockProps1 = {
        onRequestRobots: jest.fn(),
        robots: [{
            id: 1,
            name: 'John',
            email: 'John@gmail.com'
        }],
        searchField: 'john',
        isPending: false
    }
    const filteredRobots = [{
        id: 1,
        name: 'John',
        email: 'John@gmail.com'
    }];
    const wrapper1 = renderer.create(<MainPage {...mockProps1} />);
    expect(wrapper1.getInstance().filteredRobots()).toEqual(filteredRobots);
})

it('filters robots correctly 2', () => {
    const mockProps2 = {
        onRequestRobots: jest.fn(),
        robots: [{
            id: 1,
            name: 'John',
            email: 'John@gmail.com'
        }],
        searchField: 'a',
        isPending: false
    }
    const filteredRobots = [];
    const wrapper2 = renderer.create(<MainPage {...mockProps2} />);
    expect(wrapper2.getInstance().filteredRobots()).toEqual(filteredRobots);
})

