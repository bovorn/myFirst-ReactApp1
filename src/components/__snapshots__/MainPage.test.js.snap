// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MainPage without crashing 1`] = `
<div
  className="tc"
>
  <div>
    <div>
      <h1
        className="f1"
      >
        RoboFriends
      </h1>
      <div>
        <button
          color="red"
          id="counter"
          onClick={[Function]}
        >
          Count: 
          0
        </button>
      </div>
    </div>
    <input
      aria-label="Search Robot"
      className="pa3 ba b--green bg-lightest-blue"
      placeholder="Search Robot..."
      style={
        Object {
          "height": "30px",
          "maxWidth": "400px",
          "width": "400px",
        }
      }
      type="search"
    />
    <div
      style={
        Object {
          "border": "5px solid rgb(0, 88, 88)",
          "height": "720px",
          "margin": "20px auto",
          "maxWidth": "1200px",
          "overflowX": "hide",
          "overflowY": "scroll",
        }
      }
    >
      <div />
    </div>
  </div>
</div>
`;
